lifecycle/src/lifecycle_service_client.cpp

```diff
@@ -0,0 +1,161 @@
+// Copyright 2016 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <memory>
+#include <string>
+
+#include "lifecycle_msgs/msg/state.hpp"
+#include "lifecycle_msgs/msg/transition.hpp"
+#include "lifecycle_msgs/srv/change_state.hpp"
+#include "lifecycle_msgs/srv/get_state.hpp"
+
+#include "rclcpp/rclcpp.hpp"
+
+// service topics for node-attached services
+static constexpr char const * lifecycle_node = "lc_talker";
+static constexpr char const * node_get_state_topic = "lc_talker__get_state";
+static constexpr char const * node_change_state_topic = "lc_talker__change_state";
```

Comment:
These last two are only used in one other place (`lifecycle_node` is used in several places), so you could remove the last two and use them in place to simplify this demo.